modules/vye/app/models/vye/user_profile.rb

```diff
@@ -40,11 +49,44 @@ class Vye::UserProfile < ApplicationRecord
   )
 
   def self.find_and_update_icn(user:)
-    return if user.blank?
+    if user.blank?
+      Rails.logger.error "#{name}: There is no user in session."
+      return
+    end
+
+    unless user.loa3?
+      Rails.logger.error "#{name}: The user(#{user&.user_account&.id}) in session is not LOA3."
+      return
+    end
+
+    if user.ssn.blank?
+      Rails.logger.error "#{name}: The user(#{user&.user_account&.id}) in session does not have an SSN."
```

Comment:
What will `name` evaluate to here? I don't see that variable set anywhere.

Reply:
i had it as `self.name` but rubocop removed the `self`. Basically it's the class name. I wish there was a `class_name`

Reply:
got it! Thanks!